[ENH] Add new "Getting started" docs

surchs opened the pull request:
<!-- 
Please indicate after the # which issue you're closing with this PR, if applicable.
If the PR closes multiple issues, include "closes" before each one is listed.
You can also link to other issues if necessary, e.g. "See also #1234".

https://help.github.com/articles/closing-issues-using-keywords
-->
- Closes #184 
- Close #187

<!-- 
Please give a brief overview of what has changed or been added in the PR.
This can include anything specific the maintainers should be looking for when they review the PR.
-->
Changes proposed in this pull request:

-
-

<!-- To be checked off by reviewers -->
## Checklist
_Please leave checkboxes empty for PR reviewers_

- [ ] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`) _see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING/#pull-request-guidelines) for more info)_
- [ ] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Checks pass

rmanaem commented on docs/config.md:
````diff
@@ -0,0 +1,331 @@
+
+`.env` holds environment variables needed for the `f-API` deployment.
+
+``` {.bash .annotate title=".env"}
````
Nice, didn't know about this syntax!

rmanaem commented on docs/config.md:
````diff
@@ -0,0 +1,331 @@
+NB_QUERY_PORT_HOST=3000
+```
+
+1.  When a user uses the graphical query tool to query your
+    f-API, these requests will be sent from the user's machine,
+    not from the machine hosting the query tool.
+
+    Make sure you set the `NB_API_QUERY_URL` in your `.env`
+    as it will appear to a user on their own machine 
+    - otherwise the request will fail.
````
As nice as this is, it's hard to notice/find on the docs, maybe it's better to put it in an info/warning section in the main body?

surchs replied:
ok, Alyssa also pointed out that this whole thing is a duplicate from above, so we'll scrap it

rmanaem reviewed (changes requested):
Thanks for the PR @surchs!
:cherries: Please see the comments for requested changes.

alyssadai commented on docs/maintaining.md:
```diff
@@ -0,0 +1,166 @@
+    If you forget this step, `docker compose` will use the default profile (`local_node`),
+    and will only shut down, update, or start the services for that profile.
+
+## Updating the data in your graph
```
Because this section covers actually **re-annotating and re-harmonizing** data using the annotation tool and CLI, which would be targeted to a RA with dataset expertise rather than a sysadmin, I think it makes more sense to be just renamed "Updating a harmonized dataset" and moved to the "Annotating/harmonizing a dataset" section (also because at this point, we have not mentioned those tools at all).

The part I would leave on this page, under "Updating the data in your graph", is the section currently under the "Maintaining a node" page describing using `add_data_to_graph` to reupload datasets. Again, b/c this is something to be handled by the sysadmin.

Another reason I think this makes sense is because re-annotating and re-running the CLI on data may likely happen (and should be easy to do) asynchronously from when that data is actually pushed to the graph.

alyssadai reviewed (changes requested):
Hey @surchs, thanks so much for addressing this big docs reconsolidation issue! 🎉

My review mainly includes suggestions to help ensure we don't lose clarity about what tools are being deployed, what they mean/do, and that we don't muddy the waters too much for instructions meant for a sysadmin vs. other users 🤞, just to make future docs updates hopefully not so massive.

Some overarching comments:
- Linking here our suggested reordering of pages https://hackmd.io/@neurobagel/ByFWsKMMR to follow 3-ish main super-sections. With this, I think we can have the top-most "Getting started" page be just a few sentences directing readers to appropriate sections based on what they're looking to do
  - happy to help with the reordering!
- Not sure if I missed it, but I think we may have lost our [disclaimer](https://neurobagel.org/federate/#when-not-to-use-local-query-federation) on how local federation setup is not needed if a user just wants to run federated queries over publicly accessible nodes
  - Given the suggested docs reordering above, maybe this now makes more sense under the "Running cohort queries" super-section, with just a brief mention in the deployment section? 🤔 
- From reading "Maintaining a node", I feel there is still some conflicting info around what the recommended approach is to make non-example or updated datasets appear in a running graph. Is it to use the `add_data_to_graph.sh` script (which I think you link to in the 'Updating the data in graph` section)? Or to spin down the entire stack and restart it?
  - If it's the latter, I think we should add more explicit instructions to change `NB_GRAPH_DATA` in this section, and maybe also highlight the "Resetting your GraphDB instance" section more, so that we avoid problems with the persistent GraphDB home when the graph service + setup steps are relaunched.
  - If this is too much to fix as part of this PR, we can open a separate issue for this

surchs commented on docs/config.md:
```diff
@@ -0,0 +1,331 @@
+
+## Deployment
+
+### Available services
```
I disagree. The detailed docs here are linked from the getting started page. But for the first deployment, I don't really care that I'm launching a graph database and all this other stuff. I just want to see things run

surchs commented on docs/config.md:
```diff
@@ -0,0 +1,331 @@
+
+### Available services
+The Neurobagel docker compose recipe includes several services
+and coordinates them to work together:
```
OK, so iiuc you would put the "human" name first and then the docker-compose "service name" second in brackets? Would you mind suggesting this change? 

> By service name in the recipe:

I'm not sure what this addition means. Do you think that helps clarify here?

surchs commented on docs/config.md:
```diff
@@ -0,0 +1,331 @@
+    deployments running and you now want to provide federation over them.
+       - `federation`
+       - `query_tool`
+4. `local_node_query`: Legacy profile. This lets you create a local node without 
```
What you are describing is what "deprecated" or "legacy" means. Maybe "deprecated" would be better here? We don't want to just make things magically disappear, so the docs here say: "this works but will go away eventually".

surchs commented on docs/config.md:
```diff
@@ -0,0 +1,331 @@
+!!! tip
+    Double check that any environment variables you have customized in `.env` are resolved with your expected values using the command `docker compose config`.
+
+### federation service specific `.env` variables
```
OK, how about we just keep `nb_local_nodes.json` as a subsection in "Environment variables"?

surchs commented on docs/config.md:
```diff
@@ -0,0 +1,331 @@
+
+!!! warning "Do not use `localhost`/`127.0.0.1` in `local_nb_nodes.json`"
+
+    If the local node API(s) you are federating over is running on the same host machine as your federation API (e.g., the URL to access the node API is http://localhost:XXXX), make sure that you replace `localhost` with `host.docker.internal` in the `"ApiURL"` for the node inside `local_nb_nodes.json`.
```
Good point

surchs commented on docs/config.md:
```diff
@@ -0,0 +1,331 @@
+
+## Manual post-launch setup
+
+The Neurobagel docker compose recipe will automatically setup and configure 
+all services for you after deployment. 
+The following steps are only documented as a reference and for advanced users.
+You should not need to do this in most cases.
```
Nice, I like that

surchs commented on docs/getting_started.md:
````diff
@@ -0,0 +1,164 @@
+```
+3. Change `NB_API_QUERY_URL` in the `.env` file 
+
+    You **must** replace the placeholder value for `NB_API_QUERY_URL`in the `.env` file
+    
+    ```bash
+    NB_API_QUERY_URL=http://XX.XX.XX.XX
+    ```
+    with the address of the machine you are going to deploy Neurobagel on.
+
+    - If you are deploying Neurobagel **for yourself** or just to try things out locally, 
+   you can use: `NB_API_QUERY_URL=http://localhost:8080` 
+    where `:8080` is the default port for the federation API.
+    - If you are deploying Neurobagel on a server for other users, 
+   you **must** use the IP (and Port) or URL that your users will access the server with. 
````
Of these, the only variable that must be changed before launching is `NB_API_QUERY_URL` (and even that we're reconsidering). The purpose of this docs page is not to give them a complete overview but to get started. So I'd say we stick to the bare minimum but explain where to find more details

surchs commented on docs/getting_started.md:
```diff
@@ -0,0 +1,164 @@
+   
+!!! info
+
+    This is the minimal configuration you need to make before you can launch Neurobagel.
+    In most cases, and especially when you are deploying Neurobagel for other users,
+    you will have to make additional configurations. 
```
This PR is not about changing the defaults. In my opinion, `full_stack` is the best profile for "getting started" because it gets you from zero to "everything runs" in 3 steps. Again: once a user has gone through this docs page they will invariably have to go more in depth in the other docs pages. The idea here is not to be exhaustive / complete but to show people how quickly they can get going. 

see e.g. https://fastapi.tiangolo.com/#example is just a piece of code with zero context.

surchs replied:
re your suggested change: I'd say we leave it at "must change" to indicate the minimal config they just ran is not meant to be production ready.

surchs commented on docs/getting_started.md:
````diff
@@ -0,0 +1,164 @@
+cp template.env .env
+cp local_nb_nodes.template.json local_nb_nodes.json
+```
+3. Change `NB_API_QUERY_URL` in the `.env` file 
````
same as below

surchs commented on docs/getting_started.md:
```diff
@@ -0,0 +1,164 @@
+
+Once you have configured your Neurobagel deployment by editing the `.env`
+and optionally the `local_nb_nodes.json` file, you can launch the Neurobagel
+tools using `docker compose` like this: 
```
Yeah, I think linking to the detailed docs here can make sense. And a rewording from "tools" to "node" makes sense. But as I mentioned on some of the other comments: the purpose of this page is not to be a technical manual, but to get someone with initial interest far enough along that they are willing to read the more detailed stuff

surchs commented on docs/overview.md:
```diff
@@ -18,15 +18,3 @@ The Neurobagel ecosystem comprises four primary tools:
-
-    Add Neurobagel figure for overview.
-
-## Getting started
```
I get what you mean but "getting started" is now pretty obviously the next step: 
![image](https://github.com/neurobagel/documentation/assets/1302022/f8bb6cfd-68b7-4388-9a9e-2032eca8f426)

So this feels like duplication. If you really think this is not clear enough, we could add an explicit link to the "getting started" section.

Sidenote: I think in a future PR we should make the "overview" page a bit prettier.

surchs replied:
ok, with the refactored docs structure I think it makes sense to bring these links back explicitly. Will mark this as resolved

surchs commented on docs/getting_started.md:
```diff
@@ -0,0 +1,164 @@
+The following sections will show you 
+at a high level how to deploy and use the 
+Neurobagel tools on your own hardware.
```
I added an overview figure back at the top of the page and highlighted what we are deploying. I agree that we should have a (very high level) explanation on what a node is here, but I think details belong in the subsequent pages.

surchs commented on docs/config.md:
````diff
@@ -0,0 +1,331 @@
+  --clear-data
+```
+
+### Updating a dataset in the graph database
````
I agree - but these are the manual steps for data updating. To the extent that we want to keep them in the docs at all, they should live in a special section. 

As an alternative we can move all of the manual stuff to hackMD for now. wdyt?

surchs commented on docs/maintaining.md:
```diff
@@ -0,0 +1,166 @@
+### Restarting services after an update
+
+Whether you have updated the Docker images, the [configuration](config.md), or the [data](#updating-the-data-in-your-graph) 
+of your Neurobagel node, you will need to restart the services to apply the changes.
```
Hhm, I see your point. But I think we should address the issue with graph incompatibilities after restart quickly in a separate issue. I'm happy to make a complete database wipe everytime we restart the full stack - and then wait 10 minutes for everything to come back to life - if that means I don't have to remember special combos of services I need to shut down/keep alive or introduce hard to debug problems.

For now I'll add a TODO flag here.

surchs commented:
Alright folks, thanks a lot for the in depth review here - I appreciate the effort. I went through the comments / suggestions and addressed most of them. The purpose of this PR (and the new docs section) to me is really to get a sysadmin from first contact to running stack in as few steps as possible - and to provide additional detail after that to go in depth. As long as we check this box, I think this PR has served its purpose. Some of this will have to go in later issues / PRs. 

There were also a couple of change requests that are in scope of the PR, but that I disagree with in the interest of keeping the "getting started" section very action oriented and simple. I added my thoughts inline, but for me it all comes down to making the first contact with a new deploying user as easy, clear, and successful as possible so we can then tell them the details when they are excited. That said, we will continue refining the style with user feedback and make sure we get it right. 

Please both take another look at and let me know if there are any final changes needed. Let's wrap this up and spread the good word!

rmanaem commented on docs/getting_started.md:
```diff
@@ -0,0 +1,160 @@
+this will:
+
+- pull the required Docker images (if you haven't pulled them before)
+- launch the containers for [the Neurobagel services](config.md#available-services
```
```suggestion
- launch the containers for [the Neurobagel services](config.md#available-services)
```

rmanaem commented on docs/getting_started.md:
```diff
@@ -0,0 +1,160 @@
+    In most cases, and especially when you are deploying Neurobagel for other users,
+    you will have to make additional configurations. 
+
+    Please refer to [our detailed documentation]() for a complete overview of 
+    configuration options.
```
Missing link

rmanaem reviewed (approved):
Please see comments for the two small issues with the links other than that :cook:

alyssadai commented on docs/public_nodes.md:
```diff
@@ -0,0 +1,36 @@
+
+## Public Neurobagel Nodes
+
+At the moment, the following public Neurobagel nodes are available:
```
🍒 For a more concise way to indicate how to select the node, how about adding this:

```suggestion
At the moment, the following public Neurobagel nodes are available 
(you can query a specific node by selecting it from the dropdown under "Neurobagel graph" in the query tool):
```

and then removing the sentence "You can query this node by selecting..." from each node's bullet point below? 
Since the full names rather than abbreviations are used as node names anyways!

alyssadai reviewed (commented):
Thanks for all the revisions @surchs ! My final suggestions below, mostly minor formatting fixes (easier to flag those now as I read them, rather than try to remember / comb through all these pages again in future 😅) and phrasing edits for clarity.

After this, I think we're good to merge!

alyssadai reviewed (approved):
Looks good @surchs !! 

B/c I was having problems with the suggestions view, I pushed two changes 1) to fix a residual spacing problem in the environment var list (two were still appear up on the same line), and 2) to fix a broken link.

🧑‍🍳
